eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -1543,3 +1562,148 @@ def test_info_view_success(self) -> None:
         self.assertIn("version", response_data)
         self.assertIn("name", response_data)
         self.assertIn("git", response_data)
+
+
+@ddt.ddt
+# pylint: disable=too-many-ancestors
+class TestGradeAPIIntegration(
+    BaseAPIIntegrationTest,
+    UsersAPIRequestMixin,
+    EnrollmentAPIRequestMixin,
+    GradeAPIRequestMixin
+):
+    """Integration test suite for the Grade API"""
+
+    def setUp(self) -> None:
+        """Set up the test suite"""
+        super().setUp()
+        self.mode = "audit"
+        self.user = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, self.user)
+        self.create_enrollment_data = {
+            "email": self.user['email'],
+            "course_id": self.demo_course_id,
+            "mode": self.mode,
+            "is_active": True,
+            "force": True,
+        }
+        self.params = {
+            "username": self.user['username'],
+            "course_id": self.demo_course_id,
+        }
+
+    @ddt.data(
+        {"detailed": False, "grading_policy": False},
+        {"detailed": True, "grading_policy": False},
+        {"detailed": True, "grading_policy": True},
+    )
+    def test_read_valid_params_successfully(self, extra_params: dict):
+        """
+        Get grades info from a user enrolled on a course.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `get_enrollment`
+        - `get_course_grade_factory`
+        - `get_valid_course_key`
+        - `get_courseware_courses`
+        - `grade_factory`
+
+        Expected result:
+        - The status code is 200.
+        - The response includes the 'earned_grade'.
+        - The response includes the 'section_breakdown' if 'detailed' params is True, otherwise, it is not included.
+        - The response includes the 'grading_policy' if 'grading_policy' params is True, otherwise, it is not included.
+        """
+        self.create_enrollment(self.tenant_x, self.create_enrollment_data)
+        params = self.params.copy()
+        params.update(extra_params)
+
+        response = self.get_grade_info(self.tenant_x, params)
+        response_content = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn("earned_grade", response_content)
+        if extra_params["detailed"]:
+            self.assertIn("section_breakdown", response_content)
+        else:
+            self.assertNotIn("section_breakdown", response_content)
+
+        if extra_params["grading_policy"]:
+            self.assertIn("grading_policy", response_content)
+            self.assertIn("grader", response_content["grading_policy"])
+            self.assertIn("grade_cutoffs", response_content["grading_policy"])
+        else:
+            self.assertNotIn("grading_policy", response_content)
```

Comment:
Should we also check the content of the response, or is checking the existence of the keys enough?

Reply:
It seems to me that it is enough to check that the expected keys come. Evaluating whether the correct content is coming seems more related to unit tests in my opinion :thinking: what do you think? @mariajgrimaldi

Reply:
Testing each API component's behavior is part of the unit test module, but here, we're testing that all those components work well together with the Open edX definitions available. So, we should include minimum checks for the response content to ensure the integration works as expected. Do you think that would work for this specific test suite?